phpBB/assets/javascript/editor.js

```diff
@@ -24,337 +6,382 @@ function helpline(help) {
 	document.forms[form_name].helpbox.value = help_line[help];
 }
 
-/**
-* Fix a bug involving the TextRange object. From
-* http://www.frostjedi.com/terra/scripts/demo/caretBug.html
-*/ 
-function initInsertions() {
-	var doc;
-
-	if (document.forms[form_name]) {
-		doc = document;
-	} else {
-		doc = opener.document;
-	}
 
-	var textarea = doc.forms[form_name].elements[text_name];
 
-	if (is_ie && typeof(baseHeight) !== 'number') {
-		textarea.focus();
-		baseHeight = doc.selection.createRange().duplicate().boundingHeight;
+// If there's no console or any of the used console methods, just make them noop as there's nothing I can do
+window.console || (window.console = {});
+console.log || (console.log = function (){});
+console.info || (console.info = console.log);
+console.warn || (console.warn = console.log);
+console.error || (console.error = console.log);
 
-		if (!document.forms[form_name]) {
-			document.body.focus();
-		}
-	}
-}
 
-/**
-* bbstyle
-*/
-function bbstyle(bbnumber) {
-	if (bbnumber !== -1) {
-		bbfontstyle(bbtags[bbnumber], bbtags[bbnumber+1]);
-	} else {
-		insert_text('[*]');
-		document.forms[form_name].elements[text_name].focus();
-	}
-}
+var smilieBox = document.getElementById('smiley-box');
 
-/**
-* Apply bbcodes
-*/
-function bbfontstyle(bbopen, bbclose) {
-	theSelection = false;
+$(smilieBox).on('click', 'img', function (e){
+	editor.insertBBCode(e.target.alt);
+	e.preventDefault();
+});
 
-	var textarea = document.forms[form_name].elements[text_name];
 
-	textarea.focus();
+var editorConstants = {
+	'NO_VALUE' 			: -1,
+	'VALUE_IN_CONTENT' 	: -2,
 
-	if ((clientVer >= 4) && is_ie && is_win) {
-		// Get text selection
-		theSelection = document.selection.createRange().text;
+};
 
-		if (theSelection) {
-			// Add tags around selection
-			document.selection.createRange().text = bbopen + theSelection + bbclose;
-			textarea.focus();
-			theSelection = '';
-			return;
-		}
-	} else if (textarea.selectionEnd && (textarea.selectionEnd - textarea.selectionStart > 0)) {
-		mozWrap(textarea, bbopen, bbclose);
-		textarea.focus();
-		theSelection = '';
-		return;
-	}
+var editor = {
 
-	//The new position for the cursor after adding the bbcode
-	var caret_pos = getCaretPosition(textarea).start;
-	var new_pos = caret_pos + bbopen.length;
+	tokenRegex: {
+		'ALPHANUM': /^[0-9A-Za-z]+$/,
```

Comment:
You should remove them because they're not used anywhere else and good coders write consistent code.

It's also good practice and one of the defaults in JSHint.